example/extending/newlang/5-hello-python/build.mill

```diff
@@ -0,0 +1,138 @@
+// == Basic Python Pipeline 
+
+// This Example demonstrates integration of https://www.python.org[Python]
+// compilation into a Mill build to compile https://www.python.org[Python] Scripts. Mill
+// does not come bundled with https://www.python.org[Python] integration, so here we begin setting
+// one up from first principles using the Basic `Python3`(Already Present in OS) command Line
+
+
+// === Installing Python Pip
+//
+// First, we need to use the `Python3` CLI tool to install `Pip` 
+// using the official https://bootstrap.pypa.io/get-pip.py[get-pip.py] Code:
+
+package build
+import mill._
+
+def installPip: T[PathRef] = Task {
+  val pipFile = Task.dest / "get-pip.py"
+  val pipPath = os.home / ".local" / "bin" / "pip3"
+
+  if (!os.exists(pipPath)) {
+    os.call(Seq("wget", "-q", "https://bootstrap.pypa.io/get-pip.py", "-O", s"$pipFile"))
+    os.call(Seq(
+      "python3",
+      s"$pipFile",
+      "--user",
+      "--break-system-packages",
+      "--no-warn-script-location"
+    ))
+
+    os.call(Seq(
+      pipPath.toString,
+      "install",
+      "pip",
+      "--break-system-packages",
+      "--no-warn-script-location"
+    ))
+  } else {
+    println("pip is already installed, skipping installation.")
+  }
+  PathRef(Task.dest)
+}
+
+// The `installPip` task ensures the Python package manager `pip` 
+// is available in a user’s environment without requiring admin permissions.
+// It checks if `pip` is already installed in the user’s home directory. If not,
+// it downloads a setup script (`https://bootstrap.pypa.io/get-pip.py[get-pip.py]`) from Python’s official repository,
+// runs it to install `pip` specifically for the user, and updates `pip` to the latest version.
+// This allows Python packages to be managed locally without needing system-level access.
+
+// === Defining our Sources
+
+// Next, We define the `sources` and `allSources` tasks to help gather Python files for processing. 
+// `sources` sets a base directory for Python source files,
+// while `allSources` searches this directory (and its subdirectories) to collect all `.py` files.
+// Together, they enable easy access to Python files in a project,
+// making it straightforward to retrieve all relevant code files for tasks like building, testing, or analysis.
+
+def sources: T[PathRef] = Task.Source(millSourcePath / "src")
+
+def allSources: T[Seq[PathRef]] = Task {
+  os.walk(sources().path).filter(_.ext == "py").map(PathRef(_))
+}
+
+// === Compilation
+
+// The `compile` Task  ensures `pip` is installed, retrieves the main file directory, and returns it for further processing.
+// This task is considered as a part of code which is responsible for all compilation of sources and pre-requirements.
+
+def compile: T[PathRef] = Task {
+  installPip()
+  val mainFileDir = sources().path
+  
+  PathRef(mainFileDir)
+}
```

Comment:
We should perform MyPy typecheck validation here, as mentioned in the original ticket

Reply:
Maybe rename it `typecheck` instead of `compile`